cc_corpus/corpus.py

```diff
@@ -14,6 +14,7 @@
 import re
 import shutil
 
+from cc_corpus.io import BatchWriter as BatchWriterBase
```

Comment:
What does this cc_corpus.io refers to? I don't see any io.py files in the commit.

Reply:
I absolutely support the idea of using a baseclass + inheritance to sort out the mess with the BatchWriters.

But:
1. The baseclass should be named as such, BatchWriterBase instead of renaming it during an import and then shadowing its original class name. 

2. I prefer the derived classes to have unique class names, like BatchWriterJsonFlagger or such. That is because sometimes we import them in other parts of the code, and it would greatly help code clarity.

3. There is a third BatchWriter class declared in deduplication.py, please refactor that as well to use inheritance!

Reply:
1. done
2-3. won't do for this PR, I would like to keep the "status quo" as much as possible. Let's return to this later. The third BatchWriter is a bit more complicated than the other two, so it might or might not fit into the "framework" anyway.

Reply:
Okay, but we should sort out this mess one day.